packages/wallet/backend/tests/transaction/service.test.ts

```diff
@@ -0,0 +1,229 @@
+import { createContainer } from '@/createContainer'
+import { Bindings } from '@/app'
+import { env } from '@/config/env'
+import { Container } from '@/shared/container'
+import { createApp, TestApp } from '@/tests/app'
+import { Knex } from 'knex'
+import { truncateTables } from '@/tests/tables'
+import { createRequest, createResponse } from 'node-mocks-http'
+import type { AuthService } from '@/auth/service'
+import { mockedListAssets, mockedTransactionInsertObjs } from '../mocks'
+import { TransactionService } from '@/transaction/service'
+import { Transaction } from '@/transaction/model'
+import { faker } from '@faker-js/faker'
+import { loginUser } from '@/tests/utils'
+import { Account } from '@/account/model'
+import { PaymentPointer } from '@/paymentPointer/model'
+
+describe('Transaction Controller', (): void => {
+  let bindings: Container<Bindings>
+  let appContainer: TestApp
+  let knex: Knex
+  let authService: AuthService
+  let transactionService: TransactionService
+  let userId: string
+
+  // "dependency" = ALL foreign keys (paymentPointers, account, user...)
+  const prepareTransactionDependencies = async () => {
+    const account = await Account.query().insert({
+      name: faker.string.alpha(10),
+      userId,
+      assetCode: mockedListAssets[0].code,
+      assetId: mockedListAssets[0].id,
+      assetScale: mockedListAssets[0].scale,
+      virtualAccountId: 'mocked'
+    })
+
+    const paymentPointer = await PaymentPointer.query().insert({
+      url: faker.string.alpha(10),
+      publicName: faker.string.alpha(10),
+      accountId: account.id,
+      id: faker.string.uuid()
+    })
+
+    return {
+      account,
+      paymentPointer
+    }
+  }
+
+  beforeAll(async (): Promise<void> => {
+    bindings = createContainer(env)
+    appContainer = await createApp(bindings)
+    knex = appContainer.knex
+    authService = await bindings.resolve('authService')
+    transactionService = await bindings.resolve('transactionService')
+  })
+
+  beforeEach(async (): Promise<void> => {
+    const res = createResponse()
```

Comment:
request and response are not used in tests. It is enough to insert in db the fake user